### PR TITLE
Fix issue on big endian architecture. Increase jmp_buf size. Reduce OUTPUT_FORMAT hardcoding

### PR DESCRIPTION
**Pull request by cclerget**
Fix issue on big endian architecture, in `tools/lkl/lib/virtio_blk.c` capacity and sector are not converted correctly from host to little endian value.

Increase buf size from 32 to 128 in lkl_jmp_buf in order to run on arch like powerpc requiring a buf size of more than 64 long int.

More generalistic approach to compile LKL on different architecture without hardcoding value as much as possible in `arch/lkl/Makefile`, `arch/lkl/Kconfig` and `tools/lkl/Makefile.autoconf`.

---
This change is [Reviewable](https://reviewable.io/reviews/lkl/linux/476)

### CODE REVIEW

File: arch/lkl/include/asm/thread_info.h [hunk 1]
```diff
@@ -13,16 +13,16 @@ typedef struct {
...
+	unsigned long stackend;
 	mm_segment_t addr_limit;
-	struct lkl_sem *sched_sem;
-	struct lkl_jmp_buf sched_jb;
-	bool dead;
 	lkl_thread_t tid;
-	struct task_struct *prev_sched;
-	unsigned long stackend;
+	int preempt_count;
+	bool dead;
```
Comment (tavip) [inline arch/lkl/include/asm/thread_info.h]: Are these changes needed? If so can you please pull them out in a separate commit with a description about why they are needed?

Also AFAIR stackend should be the last entry in the structure.

Comment (cclerget) [reply arch/lkl/include/asm/thread_info.h]: Not needed, it was originally to minimize gcc padding by ordering members in descending order. Will remove it. Thanks !
